### PR TITLE
Build NuGet client solution with VS "15" willow

### PR DESCRIPTION
**Pull request by alpaix**
This change resolves number of issues blocking VS15 build on a box with VS
"15" willow installed at non-default location:
- Discovery of msbuild 15.1
- Microsoft.Build.\* dependency version
- VS15/VS14 facade package restore conflict
- Assembly references pointing to a default VisualStudio PublicAssemblies location
- Binding VisualStudioVersion to MSBuildToolsetVersion

Key changes:
- Added 'Debug VS14'/'Release VS14' **solution** configurations to control
  projects built for specific "toolset", i.e target VS version
- Added `HintPath` to Microsoft.Build.\* assemblies pointing to current
  `MSBuildToolsPath`
- Added `VisualStudioInstallDir` property initialized with VS install path
  from registry key
- Moved common project properties to `common.props`
- Project files cleanup

//cc @drewgil @emgarten @joelverhagen @rrelyea @rohit21agrawal

### CODE REVIEW

File: test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/NuGet.PackageManagement.UI.Test.csproj [hunk 1]
```diff
@@ -1,12 +1,12 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+  <Import Project="..\..\..\build\Common.props" Condition="Exists('..\..\..\Build\Common.props')" />
```
Comment (drewgillies) [inline test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/NuGet.PackageManagement.UI.Test.csproj:4]: $(EnlistmentRoot)? Other instances of this, but won't tag them all.

**Comment by drewgillies** [comment]
:shipit: Just a small nit, and I'd like to better understand why the core test projs are going away, but it looks good.

**Comment by alpaix** [comment]
VS14 solution build configurations exclude `NuGet.VisualStudio15.Packages` from build:

image

Whereas default Debug/Release configurations now configured with unchecked VS14 specific projects:

image

**Comment by alpaix** [comment]
@drewgil I removed number of redundant csproj files from core test projects as these projects are of project.json/xproj type.

File: test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/NuGet.PackageManagement.UI.Test.csproj (hunk 1 above)
Comment (alpaix) [reply test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/NuGet.PackageManagement.UI.Test.csproj:4]: `$(EnlistmentRoot)` is declared in `Common.props` so it's not possible to use it before this file is actually imported (chicken and egg problem?)

**Pull request merged**